numpy/core/function_base.py

```diff
@@ -91,38 +92,31 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     if dtype is None:
         dtype = dt
 
-    if num <= 0:
-        return array([], dtype)
-    if num == 1:
-        return array([start], dtype=dtype)
     y = _nx.arange(0, num, dtype=dt)
-    if endpoint:
-        num -= 1
-    y /= num
-    y *= stop - start
+
+    if num > 1:
+        delta = stop - start
+        step = delta / div
+        if step == 0:
+            # Special handling for denormal numbers, gh-5437
+            y /= div
+            y *= delta
+        else:
+            y *= step
+    else:
+        # 0 and 1 item long sequences have an undefined step
```

Comment:
Seems to me that both those cases are actually errors. Hmm, I'd even be inclined to require `num` at least two if `endpoint=True`, and at least one if not, but would be pushing things...